src/common/Analytics/AnalyticsEvents.ts

```diff
@@ -0,0 +1,70 @@
+import { EventEmitter } from 'events';
+
+export type AnalyticsEventTypeHandlers = {
+  track: (eventName: string, event: unknown) => void;
+  userPropertyUpdate: (
+    key: string,
+    value: string | number | boolean | undefined,
+  ) => void;
+  reset: () => void;
+};
+
+export type AnalyticsEventTypes = keyof AnalyticsEventTypeHandlers;
+export type AnalyticsEventTypeHandler<T extends AnalyticsEventTypes> =
+  AnalyticsEventTypeHandlers[T];
+
+export class AnalyticsEvents {
+  private emitter = new EventEmitter();
+
+  public addListener<EventType extends AnalyticsEventTypes>(
+    eventType: EventType,
+    listener: AnalyticsEventTypeHandler<EventType>,
+  ) {
+    return this.emitter.addListener(eventType, listener);
+  }
+
+  public removeListener<EventType extends AnalyticsEventTypes>(
+    eventType: EventType,
+    listener: AnalyticsEventTypeHandler<EventType>,
+  ) {
+    return this.emitter.removeListener(eventType, listener);
+  }
+
+  public emit<EventType extends AnalyticsEventTypes>(
+    eventType: EventType,
+    ...params: Parameters<AnalyticsEventTypeHandler<EventType>>
+  ) {
+    return this.emitter.emit(eventType, ...params);
+  }
+}
+
+export const analyticsEvents = new AnalyticsEvents();
```

Comment:
Should consumers be using this object, or createAnalyticsEmitter? It seems odd that we'd export both.

Reply:
The way I currently have it this is for listening and the createAnalyticsEmitter is for creating, I guess I need a listening specific one as well?